compiler-plugin/src/io/ktor/compiler/utils/ScopedExpressionEvaluator.kt

```
@@ -0,0 +1,784 @@
+        val newScope = FirEvaluationScope(variables)
+        val newTypeScope = TypeSubstitutionContext(types)
+
+        try {
+            return inScope(newScope, newTypeScope) {
+                // Create a modified expression with variables replaced by their values
+                val visitor = ScopedExpressionEvaluationVisitor(session, newScope)
+
+                // substitute variables
+                expression.accept(visitor, null)
+            }
+        } catch (e: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)